routing/cross_mwm_router.cpp

```diff
@@ -87,16 +88,16 @@ IRouter::ResultCode CalculateCrossMwmPath(TRoutingNodes const & startGraphNodes,
     return IRouter::EndPointNotFound;
 
   // Finding path through maps.
-  vector<BorderCross> tempRoad;
-  code = CalculateRoute({startNode, startNode}, {finalNode, finalNode}, roadGraph, tempRoad,
-                        delegate);
+  RoutingResult<BorderCross> tempRoad;
+  code = CalculateRoute({startNode, startNode}, {finalNode, finalNode}, roadGraph, tempRoad, delegate);
```

Comment:
Why not to return more generic routing result instead of raw cost? The API will be slightly more generic.